### PR TITLE
[3.4] Backport healthcheck code cleanup

### PR DESCRIPTION
**Pull request by siyuanfoundation**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

The same as https://github.com/etcd-io/etcd/pull/17000.
Cherry-picked necessary changes from https://github.com/etcd-io/etcd/pull/13655 to prepare for the livez/healthz backport.

### CODE REVIEW

File: tests/e2e/metrics_test.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func metricsTest(cx ctlCtx) {
 		{"/metrics", fmt.Sprintf(`etcd_server_version{server_version="%s"} 1`, version.Version)},
 		{"/metrics", fmt.Sprintf(`etcd_cluster_version{cluster_version="%s"} 1`, version.Cluster(version.Version))},
 		{"/metrics", fmt.Sprintf(`grpc_server_handled_total{grpc_code="Canceled",grpc_method="Watch",grpc_service="etcdserverpb.Watch",grpc_type="bidi_stream"} 6`)},
-		{"/health", `{"health":"true"}`},
+		{"/health", `{"health":"true","reason":""}`},
```
Comment (serathius) [inline tests/e2e/metrics_test.go:53]: Looks like you unintentionally backported a "reason" field. It's not a breaking change but also not a "code cleanup".

Comment (siyuanfoundation) [reply tests/e2e/metrics_test.go]: changed it to `json:"-"`. Would that work?

**Comment by siyuanfoundation** [comment]
CC @serathius @ahrtr

**Review by serathius — APPROVED** [review]

**Review by jmhbnz — APPROVED** [review]
Backport lgtm - Thanks @siyuanfoundation

**Review by ahrtr — APPROVED** [review]
lgtm

**Pull request merged**